src/dotnet-watch/Program.cs

```diff
@@ -121,7 +125,19 @@ private void OnCancelKeyPress(object sender, ConsoleCancelEventArgs args)
                 _reporter.Output("Shutdown requested. Press Ctrl+C again to force exit.");
             }
 
-            _cts.Cancel();
+            // always fire user interupt inline because this used to determine if the process
+            // is exiting due to user interaction or not
+            _userInterrupt.Cancel();
```

Comment:
Can you explain why there are two tokens now given that you're cancelling them one after the other? From what I can tell this is the only place either token is canceled.

Reply:
Reason for adding a separate token is to not regress on https://github.com/aspnet/DotNetTools/pull/434. Basically, when the watchee (not the watcher) exits, we want to know if it did so because the user invoked ctrl+c or a file changes. By changing to dispatching _cts.Cancel(), it appears the order in which some tasks complete changes slightly and regressed https://github.com/aspnet/DotNetTools/pull/434

https://github.com/aspnet/DotNetTools/pull/438/files#diff-7895c5cf5e852895b5c8be62b850d40fR75

Reply:
That's weird. If you're trying to determine whether or not cancelling the CTS triggered the process exit (or is that not what you're trying to determine?), wouldn't the token have to be canceled by the time the process exits in order for that to ever be true even if the cancellation was dispatched?

Reply:
good point. I played with this some more and think the original cts is good enough. Removed the duplicate cts.